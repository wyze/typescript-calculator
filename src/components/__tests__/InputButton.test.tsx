import { Input } from '../../redux/modules/operations'
import { css } from 'glamor'
import { h } from 'preact'
import InputButton from '../InputButton'
import snapshot from './utils/snapshot'

const props = {
  action: jest.fn(),
  value: '1' as Input,
}

const propsWithClassName = {
  ...props,
  className: css({ color: 'rebeccapurple' }),
}

describe('<InputButton />', () => {
  it('renders', () => snapshot(<InputButton {...props} />))
  it('renders with className', () => snapshot(<InputButton {...propsWithClassName} />))
})
