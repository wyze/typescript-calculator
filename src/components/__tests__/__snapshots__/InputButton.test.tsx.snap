// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputButton /> renders 1`] = `
<Button
  action={[Function]}
  className={undefined}
  text="1"
/>
`;

exports[`<InputButton /> renders with className 1`] = `
.glamor-0,
[data-glamor-0] {
  color: rebeccapurple;
}

<Button
  action={[Function]}
  className={
    Object {
      "data-glamor-0": "",
    }
  }
  text="1"
/>
`;
